synth/src/cli/import_utils.rs

```diff
@@ -0,0 +1,166 @@
+use crate::datasource::relational_datasource::{ColumnInfo, RelationalDataSource};
+use synth_core::{Namespace, Name, Content};
+use crate::datasource::DataSource;
+use async_std::task;
+use std::str::FromStr;
+use anyhow::{Result, Context};
+use log::debug;
+use synth_core::schema::{FieldRef, NumberContent, Id, SameAsContent, OptionalMergeStrategy, ObjectContent, ArrayContent, RangeStep, OneOfContent, VariantContent, FieldContent};
+use synth_core::schema::content::number_content::U64;
+use serde_json::Value;
+use std::convert::TryFrom;
+
+#[derive(Debug)]
+pub(crate) struct Collection {
+    pub(crate) collection: Content,
+}
+
+/// Wrapper around `FieldContent` since we cant' impl `TryFrom` on a struct in a non-owned crate
+struct FieldContentWrapper(FieldContent);
+
+pub(crate) fn build_namespace_import<T: DataSource + RelationalDataSource>(datasource: &T)
+    -> Result<Namespace> {
+    let table_names = task::block_on(datasource.get_table_names())
+        .context(format!("Failed to get table names"))?;
+
+    let mut namespace = Namespace::default();
+
+    info!("Building namespace collections...");
+    populate_namespace_collections(&mut namespace, &table_names, datasource)?;
+
+    info!("Building namespace primary keys...");
+    populate_namespace_primary_keys(&mut namespace, &table_names, datasource)?;
+
+    info!("Building namespace foreign keys...");
+    populate_namespace_foreign_keys(&mut namespace, datasource)?;
+
+    info!("Building namespace values...");
+    populate_namespace_values(&mut namespace, &table_names, datasource)?;
+
+    Ok(namespace)
+}
+
+fn populate_namespace_collections<T: DataSource + RelationalDataSource>(
+    namespace: &mut Namespace, table_names: &Vec<String>, datasource: &T) -> Result<()> {
+    for table_name in table_names.iter() {
+        info!("Building {} collection...", table_name);
+
+        let column_infos = task::block_on(datasource.get_columns_infos(table_name))?;
+
+        namespace.put_collection(
+            &Name::from_str(&table_name)?,
+            Collection::try_from((datasource, column_infos))?.collection,
+        )?;
+    }
+
+    Ok(())
+}
+
+fn populate_namespace_primary_keys<T: DataSource + RelationalDataSource>(
+    namespace: &mut Namespace, table_names: &Vec<String>, datasource: &T) -> Result<()> {
+    for table_name in table_names.iter() {
+        let primary_keys = task::block_on(datasource.get_primary_keys(table_name))?;
+
+        if primary_keys.len() > 1 {
+            bail!("Synth does not support composite primary keys")
```

Comment:
Could we perhaps have a more verbose error here like:

`bail!("{} primary keys fount at collection {}. Synth does not currently support composite primary keys.", primary_keys.len(), table_name)`